File: src/main/scala/com/digitalasset/testing/ledger/clock/TimeProvider.scala
/*
 * Copyright 2019 Digital Asset (Switzerland) GmbH and/or its affiliates
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.digitalasset.testing.ledger.clock

import java.time.Instant

import io.grpc.ManagedChannel

trait TimeProviderFactory {
  def getTimeProvider(channel: ManagedChannel, ledgerId: String): TimeProvider
}

trait TimeProvider {
  def getCurrentTime: Instant
  def setCurrentTime(time: Instant): Unit
}

File: src/main/java/com/digitalasset/testing/ledger/clock/TimeProvider.java
/*
 * Copyright 2020 Digital Asset (Switzerland) GmbH and/or its affiliates
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package com.digitalasset.testing.ledger.clock;

import java.time.Instant;

public interface TimeProvider {
  Instant getCurrentTime();

  void setCurrentTime(Instant time);
}
